### PR TITLE
start.sh: print warnings on unusual exit codes from the server process

### PR DESCRIPTION
**Pull request by dktapps**
## Introduction
This has been sitting idle on a branch for a couple of years. Not sure why it hasn't been merged as it looks basically fine as far as I can tell.

### Behavioural changes
`start.sh` will now print warnings if the server experiences any of the following errors: SIGSEGV, SIGABRT, SIGKILL, any other anomalous exit code

## Tests
Not tested, just found this while cleaning out old branches and thought it might be worth formally proposing

I tested this PR by doing the following (tick all that apply):
- [ ] Writing PHPUnit tests (commit these in the `tests/phpunit` folder)
- [ ] Playtesting using a Minecraft client (provide screenshots or a video)
- [ ] Writing a test plugin (provide the code and sample output)
- [ ] Other (provide details)

### CODE REVIEW

File: start.sh [hunk 1]
```diff
@@ -44,6 +44,27 @@ fi
...
 LOOPS=0
 
+handle_exit_code() {
+	local exitcode=$1
+	if [ "$exitcode" -eq 134 ] || [ "$exitcode" -eq 139 ]; then #SIGABRT/SIGSEGV
+		echo ""
+		echo "ERROR: The server process was killed due to a critical error (code $exitcode) which could indicate a problem with PHP."
+		echo "Updating your PHP binary is recommended."
+		echo "If this keeps happening, please open a bug report."
+		echo ""
```
Comment (ShockedPlot7560) [inline start.sh:54]: Segmentation fault which is due to plugin error could lead to a lot of bug report, no ?

Comment (dktapps) [reply start.sh:54]: PHP code shouldn't be causing segfaults

Comment (ipad54) [reply start.sh:54]: Segfault can be caused in PHP code only via third-party PHP extensions. There is an open issue with `ext-pmmpthread`, but plugin devs are not supposed to use it that way.

**Review by ipad54 — APPROVED** [review]

File: start.sh (hunk 1 above)
Comment (dktapps) [reply start.sh:54]: It's still an issue with the PHP side no matter which way you slice it.

Comment (ShockedPlot7560) [reply start.sh:54]: > PHP code shouldn't be causing segfaults

If you make some tricky code in a thread causing segfault

Comment (dktapps) [reply start.sh:54]: PHP code should not be able to cause a segfault under any conditions. If it can, that's a fault of PHP or an extension.

**Comment by dktapps** [comment]
Sorry for the spam, just testing github actions

**Pull request merged**